src/roman_datamodels/datamodels/_factory.py

```diff
@@ -0,0 +1,55 @@
+"""
+Factories for creating all the DataModel classes from RAD
+    These are used to dynamically create all the DataModels which are actually used.
+"""
+from roman_datamodels import stnode
+
+from . import _mixins
+from ._core import DataModel
+
+
+def datamodel_names():
+    """
+    A generator to grab all the datamodel names and base STNode classes from RAD
+
+    Yields
+    ------
+        node_type, datamodel_name
+    """
+    for tag in stnode._stnode.DATAMODELS_MANIFEST["tags"]:
+        schema = stnode._factories.load_schema_from_uri(tag["schema_uri"])
+        if "datamodel_name" in schema:
+            yield stnode.OBJECT_NODE_CLASSES_BY_TAG[tag["tag_uri"]], schema["datamodel_name"]
+
+
+def datamodel_factory(node_type, datamodel_name):
+    """
+    The factory for dynamically creating a DataModel class from a node_type and datamodel_name
+        Note: For DataModels requiring additional functionality, a Mixin must be added to ._mixins.py
+            with the name <datamodel_name>Mixin.
+
+    Parameters
+    ----------
+    node_type : type
+        The base STNode class to use as the base for the DataModel
+    datamodel_name : str
+        The name of the DataModel to create
+
+    Returns
+    -------
+    A DataModel object class
+    """
+    if hasattr(_mixins, mixin := f"{datamodel_name}Mixin"):
+        class_type = (DataModel, getattr(_mixins, mixin))
```

Comment:
There is a bug here, things like the 
`get_primary_array_name` cannot be properly overridden as desired.

This is because the "order" of inheritance in Python is right to left meaning it applies the right most class structure first then proceeds to the left. Thus for the `get_primary_array_name` example the Mixin will load its version and then it will be overloaded by the DataModel object. This is the reverse of what we want.

This should not effect anything added by the Mixin (as one would normally do), but since this is the way I am dynamically injecting alterations to dynamically created objects, we need to pay attention to this ordering.